File: ecommerce-api-legacy/src/middleware/error-handler.js
const { AppError } = require("../errors");

function errorHandler(error, req, res, next) {
  if (error instanceof AppError) {
    res.status(error.statusCode).send(error.message);
    return;
  }

  res.status(500).send("Erro interno");
}

module.exports = { errorHandler };

File: task-manager-api/middleware/error_handler.py
from flask import jsonify

from errors import AppError


def register_error_handlers(app):
    @app.errorhandler(AppError)
    def handle_app_error(error):
        return jsonify({"error": error.message}), error.status_code

    @app.errorhandler(Exception)
    def handle_unexpected_error(error):
        return jsonify({"error": "Erro interno"}), 500
